packages/admin/src/stack/StackLink.tsx

```diff
@@ -0,0 +1,30 @@
+import { Link } from "@material-ui/core";
+import React, { PropsWithChildren } from "react";
+import { Link as RouterLink } from "react-router-dom";
+
+import { IStackSwitchApi, useStackSwitchApi } from "./Switch";
+
+interface StackLinkProps {
+    pageName: string;
+    payload: string;
+    subUrl?: string;
+    switchApi?: IStackSwitchApi;
+}
+
+export const StackLink = ({
+    pageName,
+    payload,
+    subUrl,
+    switchApi: externalSwitchApi,
+    children,
+}: PropsWithChildren<StackLinkProps>): React.ReactElement => {
+    const internalSwitchApi = useStackSwitchApi();
+    // external switchApi allows the creation of StackLinks outside of the stack with the useStackSwitch() hook
```

Comment:
do you have a real use case for that?

Reply:
I tried to rebuild the examples in the stories with the StackLink. Here I couldn't do it, because the button is outside StackSwitch:

```tsx
// Original
const [StackSwitch, switchApi] = useStackSwitch();
return (
    <div>
        <button
            onClick={() => {
                switchApi.activatePage("page2", "foo");
            }}
        >
            page2
        </button>
        <Stack topLevelTitle="Stack">
            <StackBreadcrumbs />
            <StackSwitch>
                <StackPage name="page1">
                    <div>page1</div>
                </StackPage>
                <StackPage name="page2">
                    <div>page2</div>
                </StackPage>
            </StackSwitch>
        </Stack>
    </div>
);
```

```tsx
// With StackLink
const [StackSwitch, switchApi] = useStackSwitch();
return (
    <div>
        {/*This StackLink has is outside StackSwitch => has no StackSwitchContext => doesn't work*/}
        <StackLink pageName="page2" payload="foo">
             link to page2
        </StackLink>
        <Stack topLevelTitle="Stack">
            <StackBreadcrumbs />
            <StackSwitch>
                <StackPage name="page1">
                    <div>page1</div>
                </StackPage>
                <StackPage name="page2">
                    <div>page2</div>
                </StackPage>
            </StackSwitch>
        </Stack>
    </div>
);
```

Therefore I added the `switchApi` prop, so that it's possible to combine the `StackLink` with `useStackSwitch`:

```
<StackLink pageName="page2" payload="foo" switchApi={switchApi}>
    link to page2
</StackLink>
```

Reply:
I haven't yet looked if there is a specific case in Comet where this is needed. I can remove it for now and make an extra PR if it's needed

Reply:
yes, makes sense.

I also don't know if there is a real use case for that, but on the other side it won't do any harm or make the api ugly